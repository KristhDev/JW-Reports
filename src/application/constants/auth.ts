import { ItemOption, Precursor } from '@infrasturcture/interfaces';

/**
 * The precursors options to select
 */
export const PRECURSORS_OPTIONS: ItemOption[] = [
    { label: 'Ninguno', value: 'ninguno' },
    { label: 'Auxiliar', value: 'auxiliar' },
    { label: 'Regular', value: 'regular' },
    { label: 'Especial', value: 'especial' }
];

export const precursors: Record<Uppercase<Precursor>, Precursor> = {
    AUXILIAR: 'auxiliar',
    ESPECIAL: 'especial',
    NINGUNO: 'ninguno',
    REGULAR: 'regular'
}

export const authMessages = {
    CONFIRM_PASSWORD_EMPTY: 'La confirmación de la contraseña es requerida.',
    EMAIL_ALREADY_REGISTERED: 'Ya existe un usuario con este correo.',
    EMAIL_EMPTY: 'El correo no puede estar vacío.',
    EMAIL_INVALID: 'Correo electrónico inválido.',
    EMAIL_UPDATE_UNCHANGED: 'Para actualizar tu correo debes cambiarlo.',
    NAME_EMPTY: 'El nombre no puede estar vacío.',
    NAME_MIN_LENGTH: 'El nombre debe tener al menos 2 caracteres.',
    PASSWORD_EMPTY: 'La contraseña no puede estar vacía.',
    PASSWORD_MIN_LENGTH: 'La contraseña debe tener al menos 6 caracteres.',
    PASSWORD_NOT_MATCH: 'Las contraseñas no coinciden.',
    PASSWORD_UPDATED: 'Ha actualizado su contraseña correctamente.',
    PROFILE_UPDATED: 'Ha actualizado su perfil correctamente.',
    SURNAME_EMPTY: 'Los apellidos no pueden estar vacíos.',
    SURNAME_MIN_LENGTH: 'Los apellidos deben tener al menos 2 caracteres.',
    UNATHENTICATED: 'Para realizar está acción debe iniciar sesión.',
    UNAUTHORIZED: 'No tiene permiso para realizar está acción.',
}

export const precursorMessages = {
    PRECURSOR_EMPTY: 'El campo precursor es requerido.',
    PRECURSOR_INVALID: 'Por favor seleccione una opción de precursor.',
}