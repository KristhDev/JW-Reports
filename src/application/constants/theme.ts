import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];