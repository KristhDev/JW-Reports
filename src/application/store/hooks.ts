import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();