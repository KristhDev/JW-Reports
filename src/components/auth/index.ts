export { CredentialsForm } from './CredentialsForm';
export { ForgotPasswordForm } from './ForgotPasswordForm';
export { LoginForm } from './LoginForm';
export { ProfileForm } from './ProfileForm';
export { RegisterForm } from './RegisterForm';