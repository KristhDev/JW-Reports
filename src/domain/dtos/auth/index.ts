export * from './sign-up.dto';
export * from './update.dto';