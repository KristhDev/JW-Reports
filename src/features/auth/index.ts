export { default as authReducer, setUser, clearAuth, setIsAuthLoading } from './slice';