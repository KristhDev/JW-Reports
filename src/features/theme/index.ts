export { default as themeReducer, setTheme } from './slice';