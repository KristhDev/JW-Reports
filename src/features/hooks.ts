import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;