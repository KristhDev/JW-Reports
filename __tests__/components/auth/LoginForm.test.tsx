import React from 'react';
import { act, render, screen, waitFor, fireEvent } from '@testing-library/react-native';

/* Components */
import { LoginForm } from '../../../src/components/auth';

/* Hooks */
import { useAuth, useStatus, useTheme } from '../../../src/hooks';

/* Theme */
import { darkColors } from '../../../src/theme';

/* Setup */
import { navigateMock } from '../../../jest.setup';

/* Mocks */
import { setErrorFormMock, signInMock } from '../../mocks';

/* Mock hooks */
jest.mock('../../../src/hooks/useAuth.ts');
jest.mock('../../../src/hooks/useStatus.ts');
jest.mock('../../../src/hooks/useTheme.ts');

describe('Test in <LoginForm /> component', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    (useAuth as jest.Mock).mockReturnValue({
        state: { isAuthLoading: false },
        signIn: signInMock
    });

    (useStatus as jest.Mock).mockReturnValue({
        setErrorForm: setErrorFormMock
    });

    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors }
    });

    it('should to match snapshot', async () => {
        await waitFor(() => {
            render(<LoginForm />);
        });

        await act(() => {
            expect(screen.toJSON()).toMatchSnapshot();
        });
    });

    it('should call setErrorForm when the form is empty or invalid', async () => {
        await waitFor(() => {
            render(<LoginForm />);
        });

        await waitFor(() => {

            /* Get touchable to submit form */
            const touchable = screen.getByTestId('button-touchable');
            fireEvent.press(touchable);

            /* Check if setErrorForm is called one time */
            expect(setErrorFormMock).toHaveBeenCalledTimes(1);
        });
    });

    it('should call signIn when the form is valid', async () => {
        await waitFor(() => {
            render(<LoginForm />);
        });

        const email = 'tester@gmail.com';
        const password = 'testerpass1234';

        await act(async() => {
            await waitFor(() => {

                /* Get text inputs to type email and password */
                const inputsText = screen.getAllByTestId('form-field-text-input');
                fireEvent(inputsText[0], 'onChangeText', email);
                fireEvent(inputsText[1], 'onChangeText', password);

                /* Get touchable to submit form */
                const touchable = screen.getByTestId('button-touchable');
                fireEvent.press(touchable);

                /* Check if signIn is called one time */
                expect(signInMock).toHaveBeenCalledTimes(1);
            });
        });
    });

    it('should call navigate of useNavigation with respective values', async () => {
        await waitFor(() => {
            render(<LoginForm />);
        });

        await act(async() => {
            await waitFor(() => {

                /* Get touchable to navigate of RegisterScreen */
                const touchableSignUp = screen.getByTestId('login-form-sign-up');
                fireEvent.press(touchableSignUp);

                /* Check if navigate is called with respective arg */
                expect(navigateMock).toHaveBeenCalledWith('RegisterScreen');

                /* Get touchable to navigate of ForgotPasswordScreen */
                const touchableForgotPass = screen.getByTestId('login-form-forgor-pass');
                fireEvent.press(touchableForgotPass);

                /* Check if navigate is called with respective arg */
                expect(navigateMock).toHaveBeenCalledWith('ForgotPasswordScreen');
            });
        });
    });

    it('should disabled button then isAuthLoading is true', async () => {

        /* Mock data of useAuth */
        (useAuth as jest.Mock).mockReturnValue({
            state: { isAuthLoading: true },
            signIn: signInMock
        });

        await waitFor(() => {
            render(<LoginForm />);
        });

        await act(async() => {
            await waitFor(() => {

                /* Get submit touchable and check if disabled */
                const touchable = screen.getByTestId('button-touchable');
                expect(touchable.props.accessibilityState.disabled).toBeTruthy();
            });
        });
    });
});