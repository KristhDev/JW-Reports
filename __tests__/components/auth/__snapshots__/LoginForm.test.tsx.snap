// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LoginForm /> component should to match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 0,
      "justifyContent": "center",
      "marginBottom": 40,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "height": 187.5,
      }
    }
  />
  <View
    style={
      {
        "display": "flex",
        "marginBottom": 16,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "marginBottom": 16,
          },
          undefined,
        ]
      }
      testID="form-field-label"
    >
      Correo:
    </Text>
    <View
      style={
        {
          "borderColor": "transparent",
          "borderRadius": 7,
          "borderWidth": 1,
        }
      }
    >
      <View
        style={
          {
            "borderColor": "#FFFFFF",
            "borderRadius": 5,
            "borderWidth": 0.5,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "transparent",
                "borderRadius": 5,
                "borderWidth": 1.5,
                "flexDirection": "row",
                "paddingRight": 10,
              },
              undefined,
            ]
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            cursorColor="#C0A7E1"
            keyboardType="email-address"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSelectionChange={[Function]}
            placeholder="Ingrese su correo"
            placeholderTextColor="#BFBFBF"
            selection={
              {
                "end": 0,
                "start": 0,
              }
            }
            selectionColor="#A0B9E2"
            style={
              [
                {
                  "color": "#FFFFFF",
                  "flex": 1,
                  "fontSize": 17,
                  "paddingLeft": 10,
                  "paddingRight": 5,
                  "textAlignVertical": "center",
                },
                undefined,
              ]
            }
            testID="form-field-text-input"
            value=""
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#BFBFBF",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "display": "flex",
        "marginBottom": 40,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "marginBottom": 16,
          },
          undefined,
        ]
      }
      testID="form-field-label"
    >
      Contraseña:
    </Text>
    <View
      style={
        {
          "borderColor": "transparent",
          "borderRadius": 7,
          "borderWidth": 1,
        }
      }
    >
      <View
        style={
          {
            "borderColor": "#FFFFFF",
            "borderRadius": 5,
            "borderWidth": 0.5,
          }
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "borderColor": "transparent",
                "borderRadius": 5,
                "borderWidth": 1.5,
                "flexDirection": "row",
                "paddingRight": 10,
              },
              undefined,
            ]
          }
        >
          <TextInput
            autoCapitalize="none"
            autoCorrect={false}
            cursorColor="#C0A7E1"
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSelectionChange={[Function]}
            placeholder="Ingrese su contraseña"
            placeholderTextColor="#BFBFBF"
            secureTextEntry={true}
            selection={
              {
                "end": 0,
                "start": 0,
              }
            }
            selectionColor="#A0B9E2"
            style={
              [
                {
                  "color": "#FFFFFF",
                  "flex": 1,
                  "fontSize": 17,
                  "paddingLeft": 10,
                  "paddingRight": 5,
                  "textAlignVertical": "center",
                },
                undefined,
              ]
            }
            testID="form-field-text-input"
            value=""
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="eye-btn-touchable"
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#BFBFBF",
                    "fontSize": 25,
                  },
                  undefined,
                  {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        false,
        [
          {
            "backgroundColor": "#C0A7E1",
            "borderRadius": 10,
          },
          undefined,
        ],
      ]
    }
    testID="button-touchable"
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "minWidth": "100%",
            "paddingHorizontal": 20,
            "paddingVertical": 9,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "color": "#121212",
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 2,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
        testID="button-text"
      >
        Ingresar
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 40,
      }
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 16,
          "marginRight": 10,
        }
      }
    >
      ¿No tienes cuenta?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="login-form-sign-up"
    >
      <Text
        style={
          {
            "color": "#A0B9E2",
            "fontSize": 16,
          }
        }
      >
        Crea una aquí
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 16,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="login-form-forgor-pass"
    >
      <Text
        style={
          {
            "color": "#A0B9E2",
            "fontSize": 16,
          }
        }
      >
        Olvide mi contraseña
      </Text>
    </View>
  </View>
</View>
`;
